### PR TITLE
[hipcub] Successful build in clean chroot

### PR DESCRIPTION
**Pull request by tpkessler**
hipcub is a header-only library, there's no libhipcub.so.
Install license file, update dependencies, add optional dependency.

### CODE REVIEW

File: hipcub/PKGBUILD [hunk 1]
```diff
@@ -2,46 +2,29 @@
 pkgname=hipcub
 pkgver=3.3.0
 pkgrel=2
-pkgdesc="Reusable software components for the rocm developers."
+pkgdesc='header-only library on top of rocPRIM or CUB'
```
Comment (acxz) [inline hipcub/PKGBUILD:5]: Can you explain why this description was changed from the official description: https://github.com/ROCmSoftwarePlatform/hipCUB

File: hipcub/PKGBUILD [hunk 2]
```diff
@@ -2,46 +2,29 @@
...
 arch=('x86_64')
 url="https://github.com/ROCmSoftwarePlatform/hipCUB"
-license=('custom:NCSAOSL')
-depends=('hcc' 'hip')
-makedepends=('cmake' 'python2' 'rocminfo' 'comgr')
-source=("https://github.com/ROCmSoftwarePlatform/hipCUB/archive/rocm-$pkgver.tar.gz")
+license=('custom')
+depends=('rocprim')
+optdepends=('cuda: CUB backend')
+makedepends=('cmake' 'git' 'hcc')
```
Comment (acxz) [inline hipcub/PKGBUILD:11]: Is cuda a compile option if so would we need to create a hipcub-rocm and a hipcub-cuda package?

File: hipcub/PKGBUILD [hunk 3]
```diff
@@ -2,46 +2,29 @@
 pkgname=hipcub
 pkgver=3.3.0
 pkgrel=2
-pkgdesc="Reusable software components for the rocm developers."
+pkgdesc='header-only library on top of rocPRIM or CUB'
 arch=('x86_64')
 url="https://github.com/ROCmSoftwarePlatform/hipCUB"
```
Comment (acxz) [inline hipcub/PKGBUILD:7]: Should the url be changed from the development page to the documentation page: https://rocm-documentation.readthedocs.io/en/latest/ROCm_Libraries/ROCm_Libraries.html#hipcub?

File: hipcub/PKGBUILD [hunk 4]
```diff
@@ -2,46 +2,29 @@
...
-  export CXX="/opt/rocm/hcc/bin/hcc"
-
-  # TODO: fix libhipcub.so, it contains references to $srcdir
-  cmake -DCMAKE_BUILD_TYPE=Release \
-        -DCMAKE_INSTALL_PREFIX=/opt/rocm/hipcub \
+  CXX=/opt/rocm/hcc/bin/hcc \
+  cmake -DCMAKE_INSTALL_PREFIX=/opt/rocm/hipcub \
         -Damd_comgr_DIR=/opt/rocm/lib/cmake/amd_comgr \
         "$srcdir/hipCUB-rocm-$pkgver"
-  make
```
Comment (acxz) [inline hipcub/PKGBUILD:35]: don't we need make here?

File: hipcub/PKGBUILD (hunk 1 above)
Comment (tpkessler) [reply hipcub/PKGBUILD:5]: I find the old description meaningless and wanted to change it to something more descriptive. It's basically the first sentence in the readme of hipcub.

File: hipcub/PKGBUILD (hunk 2 above)
Comment (tpkessler) [reply hipcub/PKGBUILD:11]: It seems so. But I don't have an nVidia GPU to test it.

File: hipcub/PKGBUILD (hunk 3 above)
Comment (tpkessler) [reply hipcub/PKGBUILD:7]: This change would apply to almost all packages of the ROCm stack. Right now, $url always points to the git repo.

File: hipcub/PKGBUILD (hunk 4 above)
Comment (tpkessler) [reply hipcub/PKGBUILD:35]: hipcub is a header-only library. If you don't want to run the tests / benchmarks, you don't need to call make.

File: hipcub/PKGBUILD (hunk 1 above)
Comment (acxz) [reply hipcub/PKGBUILD:5]: Sure

File: hipcub/PKGBUILD (hunk 2 above)
Comment (acxz) [reply hipcub/PKGBUILD:11]: hmm kk this is fine for now then

File: hipcub/PKGBUILD (hunk 3 above)
Comment (acxz) [reply hipcub/PKGBUILD:7]: I think the plan is to point to the official documentation. See: https://github.com/rocm-arch/rocm-arch/pull/72#issuecomment-604835787 and the following comment.

File: hipcub/PKGBUILD (hunk 4 above)
Comment (acxz) [reply hipcub/PKGBUILD:35]: kk I see, TIL

File: hipcub/PKGBUILD (hunk 1 above)
Comment (ranisalt) [reply hipcub/PKGBUILD:5]: Capitalize it, please. It looks better with starting capital letter.

**Pull request merged**